docs/sql/casts.md

```diff
@@ -41,3 +41,5 @@ A value of type `VARIANT` can be cast to any type and will produce a
 nullable result; this kind of cast will never fail at runtime.
 
 A value of any type can be cast to a `VARIANT` type.
+
+A cast to a `ROW` type will cast recursively each field.
```

Comment:
I'm not sure what it means to cast each field recursively. My guess is, scalars remain unchanged. Do structs get cast to `ROW`? How about maps and arrays?

Reply:
you can only cast a ROW to a ROW, or perhaps a VARIANT.

Reply:
Sorry, I still don't understand what this line means (A cast to a `ROW` type will cast recursively each field).

> you can only cast a ROW to a ROW, or perhaps a VARIANT.

do we need to document this?